osu.Game/Screens/Play/SpectatorPlayer.cs

```diff
@@ -83,6 +83,18 @@ private void userSentFrames(int userId, FrameDataBundle bundle)
                 SetGameplayStartTime(score.Replay.Frames[0].Time);
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            if (HUDOverlay.PlayerSettingsOverlay.PlaybackSettings.UserPlaybackRate.Value > 1
+                && score.Replay.Frames.Count > 0
+                && DrawableRuleset.FrameStableClock.CurrentTime >= score.Replay.Frames[^1].Time)
+            {
+                HUDOverlay.PlayerSettingsOverlay.PlaybackSettings.UserPlaybackRate.Value = 1;
+            }
```

Comment:
I haven't checked in detail, but would binding to `WaitingOnFrames` simplify this? Would imagine doing something like this:
```csharp
        protected override void LoadComplete()
        {
            base.LoadComplete();

            DrawableRuleset.FrameStableClock.WaitingOnFrames.BindValueChanged(waiting =>
            {
                if (GameplayClockContainer is MasterGameplayClockContainer master)
                {
                    if (master.UserPlaybackRate.Value > 1 && waiting.NewValue)
                        master.UserPlaybackRate.Value = 1;
                }
            }, true);
        }
```